### ISSUE TITLE
PPSSPP controller mapping not accessible to screen reader users

### ISSUE THREAD

**Issue by fireseed66** [hrydgard/ppsspp#13272]
### What happens?

Screen reader users are unable to create controller mappings to suit their needs. There is no screen reader output whatsoever in the controller mapping screen.

### What *should* happen?

Buttons in the controller mapping screen should be clearly labelled with text so that screen readers can recognize them. Screen reader users should be able to interact with buttons with the keyboard I.E. by pressing Enter on an input button and quickly pressing another key to have the input mapped to that key. I wrote a [guide to using computers without site](https://houseoffireseed.ml/guides/A%20Guide%20to%20Using%20Computers%20Without%20Sight.php) which should help you understand a bit more about screen readers and how blind people interface with computers.

## CPU
Intel Core i5-5350U clocked at 1.8GHz.
## Ram
8GB
##Operating System
Windows 8.1 x64
## PPSSPP Version
PPSSPP 1.10.3

**Comment by hrydgard** [comment]
PPSSPP uses a custom UI system which does not have screen reader support unfortunately. I didn't anticipate blind or visually impaired users when creating it, my main concern was trying to find a framework that worked for both touch, controller and mouse/keyboard cross platform, and I didn't find one so I wrote my own. 

Some blind people are able to muddle through by using the ppsspp.ini file to set up, but I recognize that it's difficult to set up controls that way.

I plan to investigate what it would take to add screen reader support, but it won't happen soon unfortunately.

**Comment by fireseed66** [comment]
I'd use the controls.ini file to remap my controls if I could understand what all those numbers meant.
Sent from Windows Live Mail 2008

**Comment by unknownbrackets** [comment]
Just to clarify: is this an issue you're only seeing on that screen?  Or are you also seeing issues on all other screens too?

The current control mapping screen is probably pretty unfriendly to screen readers, because it uses symbols like plus for some actions.  I would be surprised (in a good way) if other screens are working, even though they are more friendly to the idea of a screen reader.

If there's software that can understand most of the current UI, somehow, then this becomes a lot easier to fix.

-[Unknown]

**Comment by fireseed66** [comment]
The More Settings screen seems to be affected as well.

**Comment by devinprater** [comment]
Oh I see. I think he’s using Windows, with the menu bar system. Some of the settings are accessible with a screen reader that way. This isn’t the “main screen” type thing that a controller user would use.
Devin Prater
d.prater@me.com
Https://devinprater.github.io

**Comment by fireseed66** [comment]
I'm a keyboard user (make all the jokes you want) and I want to change the controller mapping for the keyboard as I don't like the default one.
